policy-controller/k8s/api/src/policy.rs

```diff
@@ -1,5 +1,117 @@
+pub mod authorization_policy;
+pub mod meshtls_authentication;
+pub mod network_authentication;
 pub mod server;
 pub mod server_authorization;
 
-pub use self::server::{Server, ServerSpec};
-pub use self::server_authorization::{ServerAuthorization, ServerAuthorizationSpec};
+pub use self::{
+    authorization_policy::{AuthorizationPolicy, AuthorizationPolicySpec},
+    meshtls_authentication::{MeshTLSAuthentication, MeshTLSAuthenticationSpec},
+    network_authentication::{NetworkAuthentication, NetworkAuthenticationSpec},
+    server::{Server, ServerSpec},
+    server_authorization::{ServerAuthorization, ServerAuthorizationSpec},
+};
+
+/// Targets a resource--or resource type--within a the same namespace.
+#[derive(Clone, Debug, Default, serde::Deserialize, serde::Serialize, schemars::JsonSchema)]
+pub struct TargetRef {
+    pub group: Option<String>,
+    pub kind: String,
+    pub name: Option<String>,
+    pub namespace: Option<String>,
+}
+
+impl TargetRef {
+    /// Checks whether the target references the given resource type
+    pub fn targets_kind<T>(&self) -> bool
+    where
+        T: kube::Resource,
+        T::DynamicType: Default,
+    {
+        let dt = Default::default();
+
+        let mut group = &*T::group(&dt);
```

Comment:
what does `&*` do here?

Reply:
`*` is shorthand for [`Deref`](https://doc.rust-lang.org/std/ops/trait.Deref.html). So if `T::group(&dt)` returns a `String`, then `&*T::group(&dt)` gives us a `&str` because of the [`impl Deref for String`](https://doc.rust-lang.org/src/alloc/string.rs.html#2317-2324).

See, for example: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=a2da103f76a0b8ec547a090341c5a3d6

Reply:
just for education: if `T::group(&dt)` returns a String then what's the difference between `&*` vs `.as_ref()` vs `.as_str()` for String?  Or, if there's no difference, why would you use one over the others?

Though, looking at the docs, it looks like `T::group(&dt)` actually returns a `Cow<'_, str>` which implements `Deref<str>`.  So would `Deref::deref(T::group(&dt))` be equivalent?  And why would this explicit conversion be necessary in the first place if `&Cow<'_, str>` is automatically coerced to `&str` in immutable contexts?